src/components/Settings/SettingsAddressbookShare.vue

```diff
@@ -85,32 +86,88 @@ export default {
 	data() {
 		return {
 			isLoading: false,
-			usersOrGroups: [],
-			selectedUserOrGroup: []
+			inputGiven: false,
+			usersOrGroups: []
+		}
+	},
+	computed: {
+		placeholder() {
+			return t('contacts', 'Share with users or groups')
+		},
+		noResult() {
+			return t('contacts', 'No users or groups')
 		}
 	},
 	methods: {
-		limitText(count) {
-			return `and ${count} other users or groups`
+		/**
+		 * Share addressbook
+		 *
+		 * @param {Object} chosenUserOrGroup
+		 */
+		shareAddressbook({ sharee, id, group }) {
+			let addressbook = this.addressbook
+			this.$store.dispatch('shareAddressbook', { addressbook, sharee, id, group })
 		},
-
-		/* example :OC.linkToOCS('cloud', 2)+ 'groups?search=Test' */
+		/**
+		 * Format responses from axios.all and add them to the option array
+		 *
+		 * @param {Array} matches Array of matches returned from the axios request
+		 * @param {String} query
+		 * @param {Boolean} group
+		 */
+		formatMatchResults(matches, query, group) {
+			if (matches.length < 1) {
+				return
+			}
+			let regex = new RegExp(query, 'i')
+			let existingSharees = this.addressbook.shares.map(share => share.id + share.group)
+			matches = matches.filter(share => existingSharees.indexOf(share.id + group) === -1)
+			for (let i = 0; i < matches.length; i++) {
+				let matchResult = matches[i].displayname.split(regex)
+				let newMatch = {
+					sharee: matches[i].displayname,
+					id: matches[i].id,
+					matchstart: matchResult[0],
+					matchpattern: matches[i].displayname.match(regex)[0],
```

Comment:
match will return undefined if no match. so undefined[0] will throw an error :)

Reply:
so I have done a check that match has a length and if not just returned out of it